sklearn/decomposition/tests/test_truncated_svd.py

```diff
@@ -191,3 +191,16 @@ def test_truncated_svd_eq_pca(X_sparse):
     assert_allclose(Xt_svd, Xt_pca, rtol=1e-9)
     assert_allclose(pca.mean_, 0, atol=1e-9)
     assert_allclose(svd.components_, pca.components_)
+
+
+@pytest.mark.parametrize("algorithm, tol", [
+    ('randomized', 0.), ('arpack', 1e-6), ('arpack', 0.)])
```

Comment:
Why arpack twice with different `tol`?

Reply:
The patch solves the issue for 2 cases: 
- algorithm='randomized'
- algorithm='arpack' and tol != 0
I have also added the remaining case:
- algorithm='arpack' and tol = 0
so that we are sure there is no regression after the patch.

Reply:
Fine with me.